src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,235 @@
+        device_info = get_storage_query_device_info(self, self.dmg)
+        devices = [device['uuid'] for device in device_info]
+        self.log.info("Device information")
+        self.log.info("==================")
+        for index, entry in enumerate(device_info):
+            self.log.info("Device %s:", index)
+            for key in sorted(entry):
+                self.log.info("  %s: %s", key, entry[key])
+        test_dev = devices[0]
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
+
+        # 2.
+        self.log_step("Verify that each device is in a 'NORMAL' state and its LED is 'OFF'")
+        err_dev = []
+        for device in devices:
+            if not self.verify_dev_led_state(device, 'NORMAL', 'OFF'):
+                err_dev.append(device)
+        if err_dev:
+            self.fail("#Device {} not in expected NORMAL, OFF state".format(err_dev))
+
+        # 3.
+        self.log_step("Creating a pool and container with RF and starting IOR in a thread")
+        self.add_pool(connect=False)
+        self.add_container(self.pool)
+
+        job_manager = get_job_manager(self, subprocess=None, timeout=120)
+        thread_queue = Queue()
+
+        ior_test_seq = self.params.get("ior_test_sequence", "/run/ior/iorflags/*")[0]
+        thread = []
+        kwargs = {"thread_queue": thread_queue, "job_id": 1}
+        ior_kwargs = {
+            "test": self,
+            "manager": job_manager,
+            "log": "ior_thread_write_pool_test.log",
+            "hosts": self.hostlist_clients,
+            "path": self.workdir,
+            "slots": None,
+            "group": self.server_group,
+            "pool": self.pool,
+            "container": self.container,
+            "processes": self.params.get("np", "/run/ior/client_processes/*"),
+            "ppn": self.params.get("ppn", "/run/ior/client_processes/*"),
+            "intercept": None,
+            "plugin_path": None,
+            "dfuse": None,
+            "display_space": True,
+            "fail_on_warning": False,
+            "namespace": "/run/ior/*",
+            "ior_params": {
+                "oclass": oclass,
+                "flags": self.ior_w_flags,
+                "transfer_size": ior_test_seq[0],
+                "block_size": ior_test_seq[1]
+            }
+        }
+        kwargs.update(ior_kwargs)
+        thread.append(threading.Thread(target=thread_run_ior, kwargs=kwargs))
+        # Launch the IOR thread
+        self.log.info("Start IOR thread : %s", thread)
+        thread[0].start()
+
+        # 4.
+        self.log_step(
+            "Marking the {} device as faulty and verifying it is 'EVICTED' and its "
+            "LED is 'ON'".format(test_dev))
+        if not self.check_result(self.set_device_faulty(test_dev), "EVICTED", "ON"):
+            self.fail("#Result of set_device_fault, device not in expected EVICTED, ON state")
+        # check device state after set nvme-faulty
```

Comment:
We should 'register' a clean up method for resetting the faulty device at this point:
```suggestion

        # Ensure the faulty device is reset in tearDown
        self.register_cleanup(self.reset_fault_device, device=test_dev)
        
        # check device state after set nvme-faulty
```

Then we would need to add a method to this test:
```
    def reset_fault_device(self, device):
        """Call dmg storage led identify to reset the device.

        Args:
            device (str): device to reset

        Returns:
            list: a list of any errors detected when removing the pool
        """
        error_list = []
        try:
            get_dmg_response(self, self.dmg.storage_led_identify, reset=True, ids=device)
        except TestFail as error:
            test.test_log.info("  {}".format(error))
            error_list.append("Error resetting device {}: {}".format(device, error))
        return error_list
```

Reply:
Thanks for the comment, please see the update